ddtrace/llmobs/_context.py

```diff
@@ -0,0 +1,59 @@
+import contextvars
+from typing import Optional
+from typing import Union
+
+from ddtrace._trace.context import Context
+from ddtrace._trace.provider import DefaultContextProvider
+from ddtrace._trace.span import Span
+from ddtrace.ext import SpanTypes
+
+
+ContextTypeValue = Optional[Union[Context, Span]]
+
+
+_DD_LLMOBS_CONTEXTVAR: contextvars.ContextVar[ContextTypeValue] = contextvars.ContextVar(
```

Comment:
Have you considered using `ddtrace.internal.core.ExecutionContext` for this purpose?

Reply:
I wasn't aware of `ExecutionContext` while writing this PR. It looks promising but I'd prefer to get this PR in first before I work on refactoring to reuse the existing core execution contextvar. Is that okay with you?